### ISSUE TITLE
Remove page-level client unit testing and implement e2e (integration) tests using Playwright

### ISSUE THREAD

**Issue by joshuagraber** [Police-Data-Accessibility-Project/pdap.io#203]
Part of Police-Data-Accessibility-Project/data-sources-app#248 

## Rationale
- All of the components in `design-system` have unit tests run against them on every push. Thus, unit testing components and pages here is superfluous and has diminishing returns for the labor involved.
- Instead, we want to test the application's critical flows end-to-end
- We _should_ continue testing components, utilities, and any other encapsulated, simply testable functions.
- This should be done _after_ the rest of the client work for Police-Data-Accessibility-Project/data-sources-app#248 

Here are the tasks involved. We may want to break these up into an epic later, as it's quite a lot of work.
- [ ] Remove client unit tests (will happen as a part of the search-v2 work)
- [ ] Install and configure Playwright, along with mock server and mock service worker if applicable
- [ ] Add testing fixtures
- [ ] Add e2e test for login flow
- [ ] Add e2e test for change/reset password flows
- [ ] Add e2e test for search, search results, and data source details flows
- [ ] Add e2e test for user profile
- [ ] Add e2e test for new data source request

**Comment by joshuagraber** [comment]
Adding hold label, as this issue is big enough it needs to be split into an epic.
